Funding Aave Robot for Governance V2 Automation (#268)
bgd-labs/aave-proposals: src/20230817_AaveRobotActivation/AaveRobotActivation.md

---
title: "Funding Aave Robot for Governance V2 Automation"

> author: "BGD labs"

Reviewer: ```suggestion
author: "BGD labs @bgdlabs"
```

discussions: "https://governance.aave.com/t/bgd-aave-robot-v1/13091/"
---

## Simple Summary

> This proposal seeks to fund the operational cost of Aave Robot with Link tokens from the collector contract which will help to perform automation on permissionless actions of Aave Governance v2. This proposal also reimburses BGD Labs with Link spent previously on the Aave Robot.

Reviewer: ```suggestion
This proposal seeks to fund the operational cost of Aave Robot with LINK tokens from the Aave Collector contract which will help to perform automation on permissionless actions of Aave Governance v2. This proposal also reimburses BGD Labs for the LINK spent previously on the Aave Robot.
```

## Motivation

> Given the nature of Aave Governance V2, certain actions such as `queue()`, `cancel()`, and `execute()` are completely permissionless and are needed to be called by any address when the conditions are met. Manual initiation of these actions can result in unnecessary delays and expired proposals, especially when proposers is unaware of additional steps beyond proposal submission. To address these issues, and enhance the overall effectiveness of the Aave Governance, the implementation of the Aave Robot presents a solution leveraging chainlink automation to solve it. This proposal seeks to fund the operational cost of the Aave Robot to automate key actions within Aave Governance V2, ultimately streamlining processes, reducing friction.

Reviewer: ```suggestion
Given the nature of Aave Governance V2, certain permissionless actions such as `queue()`, `cancel()`, and `execute()` need to be called by any address when the conditions are met. 

Manual initiation of these actions can result in unnecessary delays and expired proposals, especially when proposers are unaware of additional steps beyond proposal submission. To address these issues, and enhance the overall effectiveness of the Aave Governance, the implementation of the Aave Robot presents a solution leveraging Chainlink Automation to solve it.

After a period of testing by BGD Labs, this proposal seeks to fund the operational cost of the Aave Robot to automate key actions within Aave Governance V2, ultimately streamlining processes, and reducing friction.
```

## Specification

> This proposal will transfer a total of: 900 aLink (~ $5600) from the collector contract across ethereum, polygon, arbitrum, optimism networks to the short executor / bridge executor. On the short executor / bridge executor, the aLink tokens transferred will be withdrawn to the Link tokens. In case of polygon, as the Link withdrawn from the pool is not an ERC-677, so we swap it one to one using PegSwap to ERC-677 Link token. Out of all the Link tokens received, 300 Link will be used to fund the pre-registered robot using the Aave CL Robot Operator contract by calling the `refillKeeper()` method. Once the keeper is refilled, a total of 600 Link will be transferred to BGD labs for their previous spending on the Aave Robot for operational expenses.

Reviewer: ```suggestion
This proposal will transfer a total of: 900 aLink (~ $5600) from the collector contract across Ethereum, Polygon, Arbitrum, Optimism networks to the short Executor / Bridge Executor.

On the Short Executor / Bridge Executor, the aLINK tokens transferred will be withdrawn to the Link tokens. In the case of Polygon, as the LINK withdrawn from the pool is not an ERC-677, the payload swaps it one-to-one using `PegSwap` to the ERC-677 LINK token.

Out of all the LINK tokens received, 300 LINK will be used to fund the pre-registered robot using the Aave CL Robot Operator contract by calling the `refillKeeper()` method.

Once the keeper is refilled, a total of 600 Link will be transferred to BGD labs for their previous spending on the Aave Robot for operational expenses.
```

After a period of testing by BGD Labs, this proposal seeks to fund the operational cost of the Aave Robot to automate key actions within Aave Governance V2, ultimately streamlining processes, and reducing friction.

## Specification

This proposal will transfer a total of: 900 aLink (~ $5600) from the collector contract across ethereum, polygon, arbitrum, optimism networks to the short executor / bridge executor. On the short executor / bridge executor, the aLink tokens transferred will be withdrawn to the Link tokens. In case of polygon, as the Link withdrawn from the pool is not an ERC-677, so we swap it one to one using PegSwap to ERC-677 Link token. Out of all the Link tokens received, 300 Link will be used to fund the pre-registered robot using the Aave CL Robot Operator contract by calling the `refillKeeper()` method. Once the keeper is refilled, a total of 600 Link will be transferred to BGD labs for their previous spending on the Aave Robot for operational expenses.

> The AaveCLRobotOperator is a contract which is used to perform admin actions on the Aave Robot, a AaveCLRobotOperator is deployed on each network, that will allow for the DAO to have "admin" control on all the underlying robots, in order to register new ones, cancel or funding them with the Link required for execution. To simplify operational complexity the role of robot guardian will have the permissions to set the maximum gas limit of the keeper and to disable automation for certain governance proposals.

Reviewer: ```suggestion
The `AaveCLRobotOperator` is a contract used to perform admin actions on the Aave Robot. An `AaveCLRobotOperator` is deployed on each network, which will allow for the DAO to have "admin" control on all the underlying robots, in order to register new ones, cancel or fund them with the LINK required for execution.

To simplify operational complexity the role of robot guardian will have the permissions to set the maximum gas limit of the keeper and to disable automation for certain governance proposals. This non-invasive role will be held by BGD Labs, as technical service provider engaged with the community.
```
